### PR TITLE
Feature/backend edit user

### PR DESCRIPTION
**Pull request by linkouth**

### CODE REVIEW

File: backend/src/modules/users/controllers/users/users.controller.ts [hunk 1]
```diff
@@ -46,4 +49,30 @@ export class UsersController {
     }
   }
 
+  @UseGuards(AuthGuard)
+  @Patch()
```
Comment (MatvienkoDO) [inline backend/src/modules/users/controllers/users/users.controller.ts:53]: May be we should specify some address. 'edit-me'

File: backend/src/modules/users/controllers/users/users.controller.ts [hunk 2]
```diff
@@ -46,4 +49,30 @@ export class UsersController {
...
+  async editUser(
+    @Body() userDto: EditUserDto,
+    @Request() request: express.Request,
+    @Response() response: express.Response,
+  ) {
+    const userId: string = request.signedCookies[userIdCookieKey];
+    userDto.id = userId;
+    const validateResult = await this.usersService.validate(userDto);
+    if (validateResult.errors) {
+      response.status(400);
```
Comment (MatvienkoDO) [inline backend/src/modules/users/controllers/users/users.controller.ts]: HttpStatus.BAD_REQUEST

File: backend/src/modules/users/controllers/users/users.controller.ts [hunk 3]
```diff
@@ -46,4 +49,30 @@ export class UsersController {
...
+  @UseGuards(AuthGuard)
+  @Patch()
+  async editUser(
+    @Body() userDto: EditUserDto,
+    @Request() request: express.Request,
+    @Response() response: express.Response,
+  ) {
+    const userId: string = request.signedCookies[userIdCookieKey];
+    userDto.id = userId;
+    const validateResult = await this.usersService.validate(userDto);
```
Comment (MatvienkoDO) [inline backend/src/modules/users/controllers/users/users.controller.ts:61]: Validation of dtos is not business of services. If we will have enough time, we will try to insert json-schema wrapped in ts.
This is just note

File: backend/src/modules/users/controllers/users/users.controller.ts [hunk 4]
```diff
@@ -46,4 +49,30 @@ export class UsersController {
...
+    if (validateResult.errors) {
+      response.status(400);
+      response.send({
+        success: false,
+        status: 'invalid_form',
```
Comment (MatvienkoDO) [inline backend/src/modules/users/controllers/users/users.controller.ts]: too general name for error. Should be unique, short, very expressive.
Maybe 'incorrect_user_edit_data' or smth like it

File: backend/src/modules/users/controllers/users/users.controller.ts [hunk 5]
```diff
@@ -46,4 +49,30 @@ export class UsersController {
...
+        errors: validateResult.errors,
+      });
+      return;
```
Comment (MatvienkoDO) [inline backend/src/modules/users/controllers/users/users.controller.ts:70]: has 'linter' underscored it? There should be empty line before return

File: backend/src/modules/users/models/edit-user.dto.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  ) {
+    this.id = id;
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.email = email;
+    this.newPassword = newPassword;
+    this.oldPassword = oldPassword;
+    this.role = role;
+  }
+}
```
Comment (MatvienkoDO) [inline backend/src/modules/users/models/edit-user.dto.ts]: no newline at the end

File: backend/src/modules/users/models/edit-user.dto.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+export class EditUserDto {
+  id: string;
```
Comment (MatvienkoDO) [inline backend/src/modules/users/models/edit-user.dto.ts]: look at src/modules/product/product.response.ts

File: backend/src/modules/users/controllers/users/users.controller.ts (hunk 1 above)
Comment (linkouth) [reply backend/src/modules/users/controllers/users/users.controller.ts:53]: I dont think so

**Review by MatvienkoDO — APPROVED** [review]

**Pull request merged**